IBM/carbon-addons-iot-react
# Pagewizard & ProgressIndicator update

Closes #1094 #476

**Summary**

Added missing `secondaryLabel` prop to PageWizard's ProgressIndicator and fixed some styling for the ProgressIndicator in general.

**Change List (commits, features, bugs, etc)**

_PageWizard_

- Added `secondaryLabel` prop to `ProgressIndicator`
- Used native vertical prop instead of injected className for `ProgressIndicator`

_PageWizard.story_

- Added `secondaryLabel` prop to `ProgressIndicator`

_PageWizard.story.storyshot_

- Updated snapshot to reflect changes

_ProgressIndicator_

- Replaced all hard-coded `bx` with `${prefix}`
- Used native vertical prop instead of injected className

_ProgressIndicator.story_

- Changed the value "secondary label" to "Optional label" to make it clearer
- hideLabels and default stepWidth example never hid the labels - added prop to hide them and optional vertical mode knob

_ProgressIndicator.story.storyshot_

- Updated snapshot to reflect changes

_ProgressIndicator CSS_

- Fixed issue whereby `progress-line` wouldn’t adjust to custom `stepWidth` #476
- Fixed broken vertical mode in `rtl`
- Changed positioning of `step button` elements back to prior to carbon upgrade. Aligned at top rather than centre.
- Generally, making `ltr` and `rtl` perfectly mirrored. 

**Acceptance Test (how to verify the PR)**

_PageWizard_
- Go to the PageWizard story
- Click on stateful example
- You should see optional label displayed and aligned with step label
- Click on RTL in storybook addon
- You should see ProgressIndicator perfectly mirrored and optional label aligned with step label

_ProgressIndicator_
- Go to ProgressIndicator
- Click on Presentation example
- You should see a contionous progress-line and no breaks in between
- Confirm by playing around with the stepWidth number in the storybook addon
- Click on hideLabels and default stepWidth example
- You should now only see the current step label displayed
